File: ImplementsTransformPlugin/src/main/groovy/cn/mycommons/implementstransformplugin/ImplementsPlugin.java
package cn.mycommons.implementstransformplugin;

import com.android.build.gradle.AppExtension;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

/**
 * ImplementsPlugin <br/>
 * Created by xiaqiulei on 2017-05-15.
 */
public class ImplementsPlugin implements Plugin<Project> {

    @Override
    public void apply(Project project) {
        AppExtension app = project.getExtensions().getByType(AppExtension.class);
        app.registerTransform(new ImplementsTransform(project));
    }
}

File: ModulePlugin/src/main/kotlin/cn/mycommons/module_plugin/ImplementsPlugin.kt
package cn.mycommons.module_plugin

import com.android.build.gradle.AppExtension
import org.gradle.api.Plugin
import org.gradle.api.Project

/**
 * ImplementsPlugin <br></br>
 * Created by xiaqiulei on 2017-05-15.
 */
class ImplementsPlugin : Plugin<Project> {

    override fun apply(project: Project) {
        val app = project.extensions.getByType(AppExtension::class.java)
        app.registerTransform(ImplementsTransform(project))
    }
}